react-carousel-component/src/carousel.jsx

```diff
@@ -0,0 +1,97 @@
+import React from 'react';
+
+class Carousel extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      index: 0
+    };
+    this.arrowClicks = this.arrowClicks.bind(this);
+    this.renderProgressDots = this.renderProgressDots.bind(this);
+    this.clickDots = this.clickDots.bind(this);
+    this.updateCarousel = this.updateCarousel.bind(this);
+  }
+
+  componentDidMount() {
+    this.updateCarousel();
+  }
+
+  updateCarousel() {
+    const intervalId = setInterval(() => {
```

Comment:
Ok, this is very close. You should store the original interval id on `this` instead of assigning it to a variable.

```js
this.interval = setInterval(() => {
```